Index: inspectdb.py
===================================================================
--- inspectdb.py	(revision 7508)
+++ inspectdb.py	(working copy)
@@ -18,7 +18,7 @@
 
         introspection_module = get_introspection_module()
 
-        table2model = lambda table_name: table_name.title().replace('_', '')
+        table2model = lambda table_name: table_name.title().replace('_', '').replace(' ', '')
 
         cursor = connection.cursor()
         yield "# This is an auto-generated Django model module."
